xls/modules/rle_dec/rle_dec.x

```diff
@@ -94,7 +94,11 @@ pub proc RunLengthDecoder<SYMBOL_WIDTH: u32, COUNT_WIDTH: u32> {
     };
     let recv_next_symbol = (state.count == bits[COUNT_WIDTH]:0);
     let (tok, input) = recv_if(tok, input_r, recv_next_symbol, state_input);
-    let next_count = input.count - bits[COUNT_WIDTH]:1;
+    let next_count = if input.count == bits[COUNT_WIDTH]:0 {
```

Comment:
q: I don't think we have a way to test for `fail!`, should we file a feature request for `assert_fail!`?